src/js/main.js

```diff
@@ -183,4 +184,92 @@ Site.Menu = {
   }
 };
 
+Site.Countdown = {
+  $countdown: $('#countdown'),
+  $countdownDays: $('#countdown-days'),
+  $countdownHours: $('#countdown-hours'),
+  $countdownMinutes: $('#countdown-minutes'),
+  $countdownSeconds: $('#countdown-seconds'),
+  dayInSeconds: 60 * 60 * 24,
+  hoursInSeconds: 60 * 60,
+  interval: undefined,
+
+  init: function() {
+    var _this = this;
+
+    // get end time from markup
+    _this.end = _this.$countdown.data('end');
+
+    var now = moment().utc();
+    var nowUnixTimestampSeconds = now.format('X');
+
+    // calculate seconds between endtime and now
+    _this.secondsRemaining = _this.end - nowUnixTimestampSeconds;
+
+    // check if already finished, then display countdown values
+    _this.checkFinished();
+    _this.displayRemainingTime();
+
+    // setup interval to count down each second and update the display values
+    _this.interval = window.setInterval(function() {
+      _this.secondsRemaining--;
+
+      _this.checkFinished();
+      _this.displayRemainingTime();
+    }, 1000);
+
+  },
+
+  displayRemainingTime: function() {
+    var _this = this;
+
+    // each of these calculates the remaining values from the seconds remaining, then calculates the remainder of seconds to be used in the next step, then displays the values
+
+    var daysRemaining = _this.secondsRemaining / _this.dayInSeconds;
+    var daysRemainingRemainder = _this.secondsRemaining % _this.dayInSeconds;
+
+    _this.$countdownDays.text(_this.displayValueFromFloat(daysRemaining));
+
+    var hoursRemaining = daysRemainingRemainder / _this.hoursInSeconds;
+
+    _this.$countdownHours.text(_this.displayValueFromFloat(hoursRemaining));
+
+    var hoursRemainingRemainder = daysRemainingRemainder % _this.hoursInSeconds;
+
+    var minutesRemaining = hoursRemainingRemainder / 60;
+
+    _this.$countdownMinutes.text(_this.displayValueFromFloat(minutesRemaining));
+
+    var minutesRemainingRemainder = hoursRemainingRemainder % 60;
+
+    _this.$countdownSeconds.text(_this.displayValueFromFloat(minutesRemainingRemainder));
+  },
+
+  checkFinished: function() {
+    var _this = this;
+
+    // if there are no seconds remaining kill the countdown
+    if (_this.secondsRemaining <= 0) {
+      window.clearInterval(_this.interval);
+
+      // perhaps we have a better solution but for now just remove the countdown
+      $('#section-countdown').remove();
+    }
+  },
+
+  displayValueFromFloat: function(input) {
+    var _this = this;
+
+    return _this.padNumber(Math.floor(input));
+  },
+
+  // https://stackoverflow.com/questions/10073699/pad-a-number-with-leading-zeros-in-javascript#10073788
+  padNumber: function(n, width, z) {
```

Comment:
I think this function should be cleaned because we will never use `width` or `z`

Reply:
I mean I didn't cus it was from that reference link and didn't feel like working out how to consolidate it, but yh